src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -253,6 +253,18 @@ public static GeneratedCodeWorkspace CreateExistingCodeProject(string outputDire
             return new GeneratedCodeWorkspace(project);
         }
 
+        public static GeneratedCodeWorkspace CreatePreviousContractFromDll(string xmlDocumentationpath, string dllPath)
```

Comment:
It does not really make sense that we return `GeneratedCodeWorkspace` here, because we are reading the contract which is not generated (now) at all.
Considering the usage of this method in other places is always followed by `GetCompilationAsync`, how about we change this method to return a `Compilation` or a `CSharpCompilation` if you need something specified to this derived type?

Reply:
Sounds good, we can just return `CSharpCompilation`.